### PR TITLE
Change project id detection, try cli args, package json data on cwd, …

### PR DESCRIPTION
**Pull request by maboiteaspam**
…fallback to default built in.

Change git history search to look for patterns `v*.*.*` or `*.*.*` or `all commits`.
Remove io.js compatibility warning.
Remove tsml dependency, adds pkg-to-id, debug.
Add lint script (jshint).
Improve git log parsing, summary was sometimes undefined.
Add cli options
* `--quiet` do not print to process.stdout
* `--all process` all commits since beginning, instead of last tag
* `--simple` will print a simple form, without additional Markdown cruft

### CODE REVIEW

**Comment by maboiteaspam** [comment]
Hope we are good to go now. Sorry for that much of delay, that was not intended.

File: changelog-maker.js [hunk 1]
```diff
@@ -1,32 +1,49 @@
...
           configName : 'changelog-maker'
         , scopes     : ['repo']
       }
 
+const gitcmd        = 'git log --pretty=full --since="{{sincecmd}}" --until="{{untilcmd}}"'
+    , commitdatecmd = '$(git show -s --format=%cd `{{refcmd}}`)'
+    , untilcmd      = ''
+    , refcmd        = argv.a || argv.all ? 'git rev-list --max-parents=0 HEAD' : 'git rev-list --max-count=1 {{ref}}'
+    , defaultRef    = '--tags=v*.*.* 2> /dev/null ' +
+    '|| git rev-list --max-count=1 --tags=*.*.* 2> /dev/null ' +
```
Comment (rvagg) [inline changelog-maker.js]: I'd like these two lines indented by another 2 characters but am not going to hold up the PR just for that

**Comment by rvagg** [comment]
Good stuff, merged and released as 1.4.0. I've rewritten your commit summary to be a bit more [idiomatic](http://git-scm.com/book/ch5-2.html#Commit-Guidelines) i.e. max 50 chars for summary and wrapping at 72 chars for description thereafter (I hate line wrapping, but this is a fairly strong convention for git).

I'll add you as a collaborator here, just make sure you keep any changes to pull requests and give time for feedback before merging anything in.

**Comment by maboiteaspam** [comment]
Thanks Rod, i ll read that and adjust those two lines (!!)

**Pull request closed without merging**